src/_ert/forward_model_runner/forward_model_step.py

```diff
@@ -217,7 +217,9 @@ def _run(self) -> Generator[Start | Exited | Running | None]:
                 exit_code = process.wait(timeout=self.MEMORY_POLL_PERIOD)
             except TimeoutExpired:
                 potential_exited_msg = (
-                    self.handle_process_timeout_and_create_exited_msg(exit_code, proc)
+                    self.handle_process_timeout_and_create_exited_msg(
```

Comment:
I need some explanation what this function actually does?

Reply:
It handles how we should react on a timeout while polling the process status. If it returns Exited, we yield it and return, otherwise we try again. 
It should fail if the running_time goes above what is configured for the forward_model, but we currently reset the running time every time we timeout, so it never reaches this code path. 
The timeout function should kill the entire process group if it is different than its own process group, then return an exited message with an error attached.